### PR TITLE
feat(gin): Add transactions to Gin integration

### PR DESCRIPTION
**Pull request by zetaab**
add missing transactions for gin

### CODE REVIEW

**Comment by tonyo** [comment]
Hey @zetaab , thanks for the changes 👍 
Looks good; the only thing: could you add a test or two, to make sure it works as expected? 

Could be something like https://github.com/getsentry/sentry-go/blob/master/http/sentryhttp_test.go, but can be even simpler (at least to make sure it doesn't crash).

**Comment by zetaab** [comment]
@tonyo yeah I will do those, just take some days :)

**Comment by zetaab** [comment]
@tonyo sorry for copy pasting the tests, but now its almost similar than in normal http lib. This does not include transaction tests. I will continue with those tomorrow.

**Comment by zetaab** [comment]
actually it was quite easy to add transaction tests, so those are included now as well

**Comment by cleptric** [comment]
Thanks, we'll take a closer look soon. For now, please revert any dependencies bumps as we do not want to raise our base level.
Regarding Gin, you can read up on our [Security Policy](https://github.com/getsentry/sentry-go/security/policy) and why we do not force an upgrade from the SDK.

File: gin/sentrygin.go [hunk 1]
```diff
@@ -46,15 +47,33 @@ func New(options Options) gin.HandlerFunc {
...
+		sentry.ContinueFromRequest(c.Request),
+		sentry.WithTransactionSource(sentry.SourceURL),
+	}
+
+	transaction := sentry.StartTransaction(ctx,
+		fmt.Sprintf("%s %s", c.Request.Method, c.Request.URL.Path),
+		options...,
+	)
+	defer func() {
+		transaction.Status = sentry.HTTPtoSpanStatus(c.Writer.Status())
```
Comment (zetaab) [inline gin/sentrygin.go:68]: added Status and tests for that as well

**Review by tonyo — APPROVED** [review]

**Comment by tonyo** [comment]
Thanks!

**Pull request merged**